### PR TITLE
Upgrade to Sync 4.7.4

### PR DESCRIPTION
**Pull request by cmelchior**
Closes: https://github.com/realm/realm-java/issues/5715

Upgrade to latest Sync release.
Doc entry for the Huawei bug fix is here: https://github.com/realm/realm.io/pull/4662

TODO
* [x] Create FAQ entry for docs

### CODE REVIEW

File: gradle-plugin/src/test/groovy/io/realm/gradle/PluginTest.groovy [hunk 1]
```diff
@@ -142,7 +142,7 @@ class PluginTest {
         assertEquals(2, project.buildscript.repositories.size())
         try {
             assertEquals(4, project.repositories.size()) // The Android plugin adds 3 different local repos
-        } catch (Exception e) {
+        } catch (Error e) {
```
Comment (Zhuinden) [inline gradle-plugin/src/test/groovy/io/realm/gradle/PluginTest.groovy]: Maybe at this point, it should just be `Throwable`

Comment (cmelchior) [reply gradle-plugin/src/test/groovy/io/realm/gradle/PluginTest.groovy]: Yup, it should. But the code has been removed now. I found a better way to debug it :)

File: README.md [hunk 2]
```diff
@@ -77,10 +77,17 @@ You may unzip the file wherever you choose.  For macOS, a suggested location is
  * If you will be building with Android Studio, you will need to tell it to use the correct NDK.  To do this, define the variable `ndk.dir` in `realm/local.properties` and assign it the full pathname of the directory that you unzipped above.  Note that there is a `local.properites` in the root directory that is *not* the one that needs to be edited.
 
     ```
-    ndk.dir=/Users/brian/Library/Android/android-ndk-r10e/r10e
+    ndk.dir=/Users/brian/Library/Android/android-ndk-r10e
 
     ```
 
+* You also need a file called `source.propertie` to the `android-ndk-r10e` folder with the following content:
```
Comment (blagoev) [inline README.md]: source.properties ?

Comment (cmelchior) [inline README.md]:
```suggestion
* You also need a file called `source.properties` to the `android-ndk-r10e` folder with the following content:
```

**Pull request merged**